lib/mongo/socket.rb

```diff
@@ -123,6 +123,7 @@ def initialize(family)
     def read(length)
       handle_errors do
         data = read_from_socket(length)
+        raise IOError unless data.length == length
```

Comment:
I found that the auth tests were failing if I checked for length 0. There is also a note in the method description saying 
https://github.com/estolfo/mongo-ruby-driver/blob/RUBY-934-socket/lib/mongo/socket.rb#L111

I'm wondering if the auth tests (scram/cr/x509) tests are failing because the socket is mocked though.